lib/wicked/IPv6.pm

```diff
@@ -0,0 +1,133 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper module for IPv6 tests
+# Maintainer: Jose Lausuch <jalausuch@suse.com>
+
+package wicked::IPv6;
+
+use Mojo::Base 'wickedbase';
+use testapi;
+
+sub evaluate_step {
+    my ($self, $cmd, $value, $expected) = @_;
+    my $output = script_output($cmd);
+    record_info('output', "cmd: $cmd \noutput:\n$output");
+    die("$value is not expected") if ($expected == 0 && $output =~ $value);
+    die("$value is expected")     if ($expected == 1 && $output !~ $value);
+}
+
+sub setup_interface {
+    my ($self, %args) = @_;
+    die '$args{iface} is required'      unless $args{iface};
+    die '$args{ifcfg_type} is required' unless $args{ifcfg_type};
+    record_info("STEP", 'Setup ' . $args{ifcfg_type} . ' interface ' . $args{iface});
+    $self->get_from_data('wicked/' . $args{ifcfg_type} . '_address/ifcfg-eth0', '/etc/sysconfig/network/ifcfg-' . $args{iface});
+    $self->wicked_command('ifup', $args{iface});
+}
+
+sub wait_for_router_announcement {
+    my ($self, %args) = @_;
+    die '$args{iface} is required' unless $args{iface};
+    record_info("STEP", 'I let ' . $args{iface} . ' wait for a router announcement');
```

Comment:
```suggestion
    record_info("STEP", 'wait for a router announcement of ' . $args{iface});
```

Reply:
while generally I can accept idea of such code structure , I still don't see good reason for keeping this ugly BDD sentences

Reply:
I changed all the sentences to make them more generic and better good looking :)